src/SFA.DAS.EmployerCommitmentsV2.Web/Mappers/Apprentice/WhatIsTheNewStartDateViewModelMapper.cs

```diff
@@ -15,18 +16,26 @@ public WhatIsTheNewStartDateViewModelMapper(ICommitmentsApiClient client)
             _client = client;
         }
 
-        public async Task<WhatIsTheNewStartDateViewModel> Map(EmployerLedChangeOfProviderRequest source)
+        public async Task<WhatIsTheNewStartDateViewModel> Map(ChangeOfProviderRequest source)
         {
-            var apprenticeship = await _client.GetApprenticeship(source.ApprenticeshipId);
+            var apprenticeship = await _client.GetApprenticeship(source.ApprenticeshipId.Value);
             
             return new WhatIsTheNewStartDateViewModel
             {
                 AccountHashedId = source.AccountHashedId,
                 ApprenticeshipHashedId = source.ApprenticeshipHashedId,
                 ProviderName = source.ProviderName,
-                ProviderId = source.ProviderId,
+                ProviderId = source.ProviderId.Value,
                 EmployerWillAdd = source.EmployerWillAdd,
                 StopDate = apprenticeship.StopDate.Value,
+                NewStartMonth = source.NewStartMonth,
+                NewStartYear = source.NewStartYear,
+                NewEndMonth = source.NewEndMonth,
+                NewEndYear = source.NewEndYear,
+                NewEndDate = (source.NewEndMonth.HasValue && source.NewEndYear.HasValue) 
```

Comment:
The new start date mapper is different from the new end date mapper because there is additional validation when changing your answer that stops you from entering a date after the new stop date